### ISSUE TITLE
Operator setting on 0 instead of 1

### ISSUE THREAD

**Issue by andr229k** [ajxs/yamaha_dx97#3]
I don’t know, If I should define this as an end-user error, but I still like to describe the experience. A friend of mine wanted to test this and compared the sounds with his Yamaha DX-7 with importing SysEx.

He experienced that DX-9 sound different from DX-7, which was a very confusing experience. Later he found out that the first operator was set to 0 after setting all operators to 1, then all worked normal.

Question:
Are the first operators set as 0 per default by the firmware or changed to 0 per imported SysEx?

**Comment by ajxs** [comment]
I really appreciate you and your friend helping test the firmware! Thank you very much for that.
The operator On/Off status should be reset when receiving a patch over SysEx. I just confirmed this with my DX9 using the latest version of the firmware, and it works correctly for me. 
When I was testing this, I did find a bug where the operator On/Off statuses are improperly reset when moving between Edit/Play modes. I'll fix that. Thanks for helping bring this to my attention!

Something worth noting: The operator On/Off status in the DX7 is a software-only feature: The EGS/OPS chips inside the DX7 and DX9 don't actually support turning an operator off. The DX7 and DX9 firmware test the On/Off status of each voice when adding a new note. If an operator is disabled, its volume is just set to 0. 
The operator On/Off status isn't actually saved in the patch data either, only in the synth memory. I think the DX7v1.8 firmware changed how this worked. I'm also pretty sure the DX7 II changed this as well.
It's a bit of an unfortunate feature in my opinion, but it's there in the DX7 and DX9 firmware, so it's here as well.

**Comment by ajxs** [comment]
I just checked, and this behavior is in the original DX9 firmware. It resets the operator statuses when pressing the 'Play Mode' button. It's not in the DX7 firmware though. It doesn't really make much sense to me. I guess it should be removed.

**Comment by ajxs** [comment]
I was able to resolve an issue where changing between 'Play Mode', and other modes was triggering incorrect button-press events. This most likely accounted for the issue that your friend saw. In the original firmware, a button-press event is triggered when you change between 'Play' mode, and 'Edit', or 'Function' to switch back to whatever parameter you were previously looking at in that mode. There was a bug in how the Yamaha DX9/7 firmware was handling this. This has been fixed now. 
Thanks for bringing this to my attention!
